Typescript

lriggle-strib opened the pull request:
## Release Description
- Update the build process by:
    - Rebuild functionality using TypeScript instead of plain javascript (this forces the code to need a higher version of Node to successfully run)
    - Build icon map that ensures new icons don't disrupt the codepoint IDs to already existing icons
- Adding in the ability to alias an icon and be able to use both versions. This feature set includes:
    - Fixed a number of bugs that had not been previously addressed.
    - Completely takes the font generator out of building the css/scss/js/html/json files, since the generator functionality was NOT maintaining the codepoints as expected between builds.
        - This was unexpected with the original adjustments made the week of 4/3/23
    - Added in individual info pages for each icon, generated dynamically during the build process
    - Updates the names of some of the icons so they are no longer prefixed by `z#-` or `strib-` prior to being included into the font.
        - Example: `z1-account-widget` is now `account-widget` (and used as `.strib-account-widget`)
        - Example: `strib-logo` is now `logo` (and used as `.strib-logo`)
        - This also affects the variable names that were accessible as `--strib-strib-ZYX` and makes them accessible as `--strib-XYZ` for the updated versions of the icon names

lriggle-strib committed:
Part of integrating typescript into the application

 Changes to be committed:
	modified:   .gitignore
	modified:   build.js
	modified:   package-lock.json
	modified:   package.json
	new file:   source/CodepointsMap.ts
	new file:   source/Icon.ts
	new file:   source/IconsMap.ts
	new file:   source/build.ts
	new file:   source/codepoints.map.json
	modified:   source/icons.yml
	new file:   source/index.ts
	new file:   source/tsconfig.json

lriggle-strib committed:
Starting on the build process

 Changes to be committed:
	modified:   build.js
	modified:   package-lock.json
	modified:   package.json
	modified:   source/CodepointsMap.ts
	modified:   source/IconsMap.ts
	modified:   source/index.ts
	renamed:    source/templates/template.hbs.css -> source/templates/template.css.hbs
	renamed:    source/templates/template.hbs.html -> source/templates/template.html.hbs
	renamed:    source/templates/template.hbs.js -> source/templates/template.js.hbs
	renamed:    source/templates/template.hbs.json -> source/templates/template.json.hbs
	renamed:    source/templates/template.hbs.scss -> source/templates/template.scss.hbs

lriggle-strib committed:
Finish initial typescript implementation. Extraneous packages have
also been removed

 Changes to be committed:
	deleted:    build.js
	modified:   package-lock.json
	modified:   package.json
	modified:   source/CodepointsMap.ts
	modified:   source/index.ts
	modified:   source/templates/template.css.hbs
	modified:   source/templates/template.html.hbs
	modified:   source/templates/template.js.hbs
	modified:   source/templates/template.scss.hbs

lriggle-strib committed:
Add in some additional comments and document blocks. Update the
README

 Changes to be committed:
	modified:   README.md
	modified:   package-lock.json
	modified:   source/Icon.ts
	modified:   source/IconsMap.ts
	deleted:    source/build.ts
	modified:   source/index.ts

lriggle-strib committed:
Some minor improvements

 Changes to be committed:
	modified:   source/CodepointsMap.ts
	modified:   source/Icon.ts
	modified:   source/index.ts

lriggle-strib committed:
Support icon aliases as well as potentially fixing issues with
adding new icons.

 Changes to be committed:
	modified:   source/CodepointsMap.ts
	modified:   source/Icon.ts
	modified:   source/IconsMap.ts
	modified:   source/codepoints.map.json
	modified:   source/icons.yml
	new file:   source/icons/account-widget.svg
	new file:   source/icons/eedition-filled.svg
	modified:   source/icons/gift-article.svg
	modified:   source/icons/link-share-icon.svg
	renamed:    source/icons/strib-logo.svg -> source/icons/logo.svg
	new file:   source/icons/pause-audio-icon.svg
	new file:   source/icons/play-audio-icon.svg
	modified:   source/icons/save-article-filled.svg
	deleted:    source/icons/save-article-open.svg
	modified:   source/icons/save-article.svg
	modified:   source/icons/share-icon.svg
	modified:   source/icons/sms-text-share-icon.svg
	new file:   source/icons/social-tiktok-circle.svg
	new file:   source/icons/social-tiktok.svg
	modified:   source/icons/social-tumblr-circle.svg
	new file:   source/icons/star.svg
	deleted:    source/icons/strib-star.svg
	modified:   source/icons/undo-arrow.svg
	deleted:    source/icons/z1-account-widget.svg
	deleted:    source/icons/z1-pause-audio-icon.svg
	deleted:    source/icons/z1-play-audio-icon.svg
	deleted:    source/icons/z1-social-tiktok-circle.svg
	deleted:    source/icons/z1-social-tiktok.svg
	deleted:    source/icons/z2-eedition-filled.svg
	modified:   source/index.ts
	modified:   source/templates/template.css.hbs
	modified:   source/templates/template.html.hbs

lriggle-strib committed:
Make sure to include the strib prefix on the alias classes, that
way it matches the svg stuff down below

 Changes to be committed:
	modified:   source/templates/template.html.hbs

lriggle-strib committed:
Make some adjustments to the index.html template, and add an
additional template to make an icon-specific page

 Changes to be committed:
	modified:   source/IconsMap.ts
	modified:   source/icons.yml
	modified:   source/icons/account-widget.svg
	modified:   source/icons/logo.svg
	modified:   source/index.ts
	new file:   source/templates/icon.html.hbs
	new file:   source/templates/index.html.hbs
	deleted:    source/templates/template.html.hbs

lriggle-strib committed:
move things around and update some styles

 Changes to be committed:
	modified:   source/templates/icon.html.hbs
	modified:   source/templates/index.html.hbs

lriggle-strib committed:
Some additional adjustments

 Changes to be committed:
	modified:   source/templates/icon.html.hbs

lriggle-strib committed:
Bufixes for some of the template files

 Changes to be committed:
	modified:   source/templates/icon.html.hbs
	modified:   source/templates/template.css.hbs
	modified:   source/templates/template.scss.hbs

lriggle-strib committed:
Ensure that we only try and process image files from the icons
directory.

 Changes to be committed:
	modified:   source/IconsMap.ts

lriggle-strib committed:
Try and enforce the node version when running commands. Also just
try and clean up the styles on the icon pages

 Changes to be committed:
	modified:   .gitignore
	new file:   .npmrc
	modified:   README.md
	modified:   package-lock.json
	modified:   package.json
	modified:   source/index.ts
	modified:   source/templates/icon.html.hbs

lriggle-strib committed:
Remove this console.log line because it was left on accident and
causes errors

 Changes to be committed:
	modified:   source/templates/template.js.hbs

lriggle-strib committed:
Put a var back into the account-widget icon. It was accidentally
removed. Set the logo-text variable to not have a default color,
to mimic the behavior of the existing icon (but the var is still
usable). Don't write the icon map to a file anymore.

 Changes to be committed:
	modified:   source/IconsMap.ts
	modified:   source/icons/account-widget.svg
	modified:   source/icons/logo.svg
	modified:   source/index.ts

lriggle-strib committed:
Make sure currentColor is used as the fill color for the SVG icons
this will make styling and theming easiser.

 Changes to be committed:
	modified:   source/templates/template.css.hbs
	modified:   source/templates/template.scss.hbs

lriggle-strib committed:
Increment the version # to 14, revamp the documentation on the
index page, and make sure all icons are listed in the yml file

 Changes to be committed:
	modified:   package.json
	modified:   source/icons.yml
	modified:   source/templates/icon.html.hbs
	modified:   source/templates/index.html.hbs

stribJamieHutt reviewed (commented):
The Readme makes sense can seems to cover the bases. Two things I noticed:
- I think "npm run serve" should be "npm run server"
- "http://lcoalhost:3000" should be "http://localhost:3000"

lriggle-strib committed:
Fix typos!

 Changes to be committed:
	modified:   README.md
	modified:   source/templates/index.html.hbs

lriggle-strib committed:
Make sure to include an example of @import usage

 Changes to be committed:
	modified:   source/templates/index.html.hbs

lriggle-strib committed:
Adding a new icon doesn't have to generate a new map file.

 Changes to be committed:
	modified:   source/CodepointsMap.ts

lriggle-strib committed:
Fix bugs when adding multiple icons at the same time

 Changes to be committed:
	modified:   README.md
	modified:   source/Icon.ts
	modified:   source/IconsMap.ts
